### PR TITLE
python3Packages.json-stream: 1.5.1 -> 2.1.1

### PR DESCRIPTION
**Pull request by winterqt**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winterqt** [comment]
@fabaff Any interest in maintaining the new Rust-based tokenizer?

File: pkgs/development/python-modules/json-stream-rs-tokenizer/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  doCheck = false; # tests require json-stream -- technically we could fall back to its pure python impl to run these tests, but meh.
+
+  meta = with lib; {
+    description = "A faster tokenizer for the json-stream Python library";
+    homepage = "https://github.com/smheidrich/py-json-stream-rs-tokenizer";
+    license = licenses.mit;
+    maintainers = with maintainers; [ winter ];
+  };
+}
```
Comment (mweinelt) [inline pkgs/development/python-modules/json-stream-rs-tokenizer/default.nix:40]: Infinite recursion escape.

```suggestion
let self = buildPythonPackage rec {
  pname = "json-stream-rs-tokenizer";
  version = "0.4.13";
  format = "setuptools";

  src = fetchFromGitHub {
    owner = "smheidrich";
    repo = "py-json-stream-rs-tokenizer";
    rev = "v${version}";
    hash = "sha256-9pJi80V7WKvsgtp0ffItWnjoOvFvfE/Sz6y2VlsU+wQ=";
  };

  postPatch = ''
    cp ${./Cargo.lock} ./Cargo.lock
  '';

  cargoDeps = rustPlatform.fetchCargoTarball {
    inherit src postPatch;
    name = "${pname}-${version}";
    hash = "sha256-TjRdHSXHmF6fzCshX1I4Sq+A/fEmBHDPGZvJUxL13aM=";
  };

  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [ cargoSetupHook rust.cargo rust.rustc ]);

  doCheck = false; # infinite recursion with json-stream
  
  passthru.tests = {
    doCheck = self.overridePythonAttrs (_: { doCheck = true; });
   };

  meta = with lib; {
    description = "A faster tokenizer for the json-stream Python library";
    homepage = "https://github.com/smheidrich/py-json-stream-rs-tokenizer";
    license = licenses.mit;
    maintainers = with maintainers; [ winter ];
  };
};
in self
```

**Comment by winterqt** [comment]
@ofborg build python3Packages.json-stream python3Packages.json-stream-rs-tokenizer.tests

**Comment by winterqt** [comment]
Need to fix tests (they weren't running before, and they depend on 2 unpackaged dependencies).

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 210141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>mitmproxy2swagger</li>
    <li>python310Packages.json-stream</li>
    <li>python310Packages.json-stream-rs-tokenizer</li>
    <li>python311Packages.json-stream</li>
    <li>python311Packages.json-stream-rs-tokenizer</li>
  </ul>
</details>

**Comment by fabaff** [comment]
https://github.com/ofborg build python3Packages.json-stream python3Packages.json-stream-rs-tokenizer.tests

File: pkgs/development/python-modules/json-stream/default.nix [hunk 2]
```diff
@@ -25,6 +26,7 @@ buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/json-stream/default.nix:26]:
```suggestion

  buildInputs = [
    iconv
  ];

```
starts to fix the darwin build failure

**Comment by fabaff** [comment]
@ofborg build python310Packages.json-stream python311Packages.json-stream python310Packages.json-stream-rs-tokenizer.tests

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 210141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>mitmproxy2swagger</li>
    <li>python310Packages.json-stream</li>
    <li>python310Packages.json-stream-rs-tokenizer</li>
    <li>python311Packages.json-stream</li>
    <li>python311Packages.json-stream-rs-tokenizer</li>
  </ul>
</details>

**Comment by fabaff** [comment]
For now I can't do much for for Darwin.

**Review by fabaff — APPROVED** [review]
```bash
...ew/pr-210141-5/results/python311Packages.json-stream/lib/python3.11/site-packages]$ python
Python 3.10.10 (main, Feb  7 2023, 12:19:31) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json_stream
>>> def some_iterator():
...     yield b'{"some":'
...     yield b' "JSON"}'
... 
>>> data = json_stream.load(some_iterator())
>>> assert data['some'] == "JSON"
````

**Pull request merged**